File: src/leetcode/2026/04/Solution766.java
public class Solution766 {
    public boolean isToeplitzMatrix(int[][] matrix) {
        int rows = matrix.length;
        int cols = matrix[0].length;
        for (int i = 0; i < rows - 1; i++) {
            for (int j = 0; j < cols - 1; j++) {
                if (matrix[i][j] != matrix[i + 1][j + 1]) {
                    return false;
                }
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Solution766 solution = new Solution766();

        int[][] matrix = { { 1, 2, 3 }, { 5, 1, 2 }, { 9, 5, 1 } };
        boolean output = solution.isToeplitzMatrix(matrix);
        System.out.println("Output: " + output);

        matrix = new int[][] { { 1, 2 }, { 2, 2 } };
        output = solution.isToeplitzMatrix(matrix);
        System.out.println("Output: " + output);
    }
}

File: src/leetcode/2026/04/Solution766.cpp
#include <vector>

using namespace std;

class Solution766 {
public:
    bool isToeplitzMatrix(vector<vector<int>>& matrix) {
        int m = matrix.size();
        int n = matrix[0].size();
        for (int i = 0; i < m - 1; ++i) {
            for (int j = 0; j < n - 1; ++j) {
                if (matrix[i][j] != matrix[i + 1][j + 1]) {
                    return false;
                }
            }
        }
        return true;
    }
};

int main() {
    Solution766 solution;

    vector<vector<int>> matrix1 = {
        {1, 2, 3, 4},
        {5, 1, 2, 3},
        {9, 5, 1, 2}
    };
    bool result1 = solution.isToeplitzMatrix(matrix1);
    printf("Result 1: %s\n", result1 ? "true" : "false");

     vector<vector<int>> matrix2 = {
        {1, 2},
        {2, 2}
    };
    bool result2 = solution.isToeplitzMatrix(matrix2);
    printf("Result 2: %s\n", result2 ? "true" : "false");
    
    return 0;
}